parser-typechecker/src/Unison/Runtime/Interface.hs

```diff
@@ -195,23 +198,26 @@ allocType ctx r cons =
   pure $ ctx {dspec = Map.insert r cons $ dspec ctx}
 
 recursiveDeclDeps ::
-  Set RF.LabeledDependency ->
   CodeLookup Symbol IO () ->
   Decl Symbol () ->
   -- (type deps, term deps)
-  IO (Set Reference, Set Reference)
-recursiveDeclDeps seen0 cl d = do
-  rec <- for (toList newDeps) $ \case
-    RF.DerivedId i ->
-      getTypeDeclaration cl i >>= \case
-        Just d -> recursiveDeclDeps seen cl d
```

Comment:
Here we should have been collecting the 'seen' results from this recursive call to pass on to the next iteration of the for loop. The state monad now handles this for us.